### DISCUSSION TITLE
Document a Typescript Interface in MDX Format

### DISCUSSION THREAD

**Discussion by dkleber89** [storybookjs/storybook#14995] [Help] [answered]
Hi,

we build a React Library for our Company so every department with frontend Projects can consume this lib. In this Lib we have also many "non visible" components like hooks, stores, contexts.

For this components we use the mdx format to document how the works and how they should be implemented. Is there a way to document the typescript interfaces automatically like it works on components for the component props?

At this time we do it manually with a normal markdown table. the problem is when some changes made on the interface often the documentation stays old.

Would be happy when someone can give me a hint how i can this solved.

Thanks a lot

**Comment by shilman** [comment, marked as answer]
This is a commonly-requested feature, and I've documented it here https://github.com/storybookjs/storybook/issues/15008

Unfortunately, it's not currently on the roadmap, but I've marked it as `todo` so that it doesn't get lost.
